document-readers/tika-reader/src/main/java/org/springframework/ai/reader/tika/TikaDocumentReader.java

```diff
@@ -143,7 +149,11 @@ public TikaDocumentReader(Resource resource, ContentHandler contentHandler, Extr
 	public List<Document> get() {
 		try (InputStream stream = this.resource.getInputStream()) {
 			this.parser.parse(stream, this.handler, this.metadata, this.context);
-			return List.of(toDocument(this.handler.toString()));
+			return textSplitter.split(this.handler.toString(), 800)
```

Comment:
we shouldn't hard code a value here, but make it configurable.  It would be nice if the Tika document reader was 'smarter' to break things up.

Reply:
Fair enough, but...I got that hard-coded number from `TokenTextSplitter` itself where it's hardcoded there in the `splitText()` method. But sure, I'll make it configurable (probably tomorrow).